### PR TITLE
add list_{container,image} to the API

### PR DESCRIPTION
**Pull request by TomasTomecek**
ready for review

### CODE REVIEW

**Comment by TomasTomecek** [comment]
@jscotka thanks for review, I still need to finish the nspawn part

**Comment by TomasTomecek** [comment]
We agreed to split this PR to two:
1. This PR, define list_* API and implement it for docker backend
2. The new PR: implement for nspawn

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/CsILg](https://da.gd/CsILg)__
* Comment: __See details for logs__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/kMOu](https://da.gd/kMOu)__
* Comment: __See details for logs__

File: conu/backend/docker/backend.py [hunk 1]
```diff
@@ -21,14 +21,33 @@
...
 
 logger = logging.getLogger(__name__)
 
 
+def parse_reference(reference):
+    """
+    parse provided image reference into <image_repository>:<tag>
+
+    :param reference: str, e.g. (registry.fedoraproject.org/fedora:27)
+    :return: tuple, ("registry.fedoraproject.org/fedora", "27")
```
[the remark this answers is not in the document]
Comment (TomasTomecek) [reply conu/backend/docker/backend.py]: nice catch; changed to tuple

**Comment by TomasTomecek** [comment]
merge time?

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/o68PH](https://da.gd/o68PH)__
* Comment: __See details for logs__

File: tests/integration/test_docker.py [hunk 2]
```diff
@@ -338,3 +349,13 @@ def test_run_with_volumes_metadata_check():
         assert mount["Mode"] == "Z"
 
         container.delete(force=True)
+
+
+def test_list_containers():
```
Comment (dhodovsk) [inline tests/integration/test_docker.py]: nit: there could be another check - add new container and check len is incremented

Comment (TomasTomecek) [reply tests/integration/test_docker.py]: on it

Comment (TomasTomecek) [reply tests/integration/test_docker.py]: updated

**Comment by TomasTomecek** [comment]
\o/

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/pqe0F](https://da.gd/pqe0F)__
* Comment: __See details for logs__

**Pull request merged**